[Term Entry] Python: Numpy ndarray.astype() (#4881)
Codecademy/docs: content/numpy/concepts/ndarray/terms/astype/astype.md

---
Title: '.astype()'
Description: 'Converts data types of ndarray elements.'
Subjects:
  - 'Computer Science'
  - 'Data Science'
  - 'Web Development'
Tags:
  - 'Data Structures'
  - 'Arrays'
  - 'Functions'
  - 'NumPy'
CatalogContent:
  - 'learn-python-3'
  - 'paths/data-science'
---

The **`.astype()`** function in NumPy allows changing the data type of elements in an array. This is useful for ensuring data is in the correct format, such as converting floating-point numbers to integers or changing integers to strings.

## Syntax

```py
ndarray.astype(dtype, order='K', casting='unsafe', subok=True, copy=True)
```

## Parameters

> - **dtype**: The new data type for the array elements (e.g., **int**, **float**, **str**).
> - **order**: Specifies the memory layout order:
>   - **`'C'`** for row-major (C-style) order.
>   - **`'F'`** for column-major (Fortran-style) order.
>   - **`'A'`** (default) or 'K' to keep the current order.
> - **casting**: Controls allowed data conversions:
>   - **'no'** to prevent any conversion.
>   - **'equiv'** to allow only byte-order changes.
>   - **'safe'** to ensure no data loss during conversion.
>   - **'same_kind'** to allow safe conversions within similar types.
>   - **'unsafe'** (default) to allow any conversion, even if data might be lost.
> - **subok**: If **`True`** (default), subclasses of the array will be preserved. If **`False`**, the result will always be a base-class array.
> - **copy**: If **`True`** (default), always makes a new copy of the array. If **`False`**, reuses the input array if possible.

Reviewer: ```suggestion
- `dtype`: The desired data type for the array elements.
- `order`: Specifies the memory layout order. It can take the following values:
  - `C`: C-style (row-major) order.
  - `F`: Fortran-style (column-major) order.
  - `A`: When A is specified, NumPy prioritizes Fortran ('F') order if all arrays are Fortran contiguous; otherwise, it defaults to 'C' order.
  - `K`: This is the default value, it keeps the order of the input array.
- `casting`: Specifies how casting should be handled if the desired data type is different from the current data type. It can take the following values:
  - `no`: The data type will not be cast.
  - `equiv`: To allow only byte-order changes.
  - `safe`: To ensure no data loss during conversion.
  - `same_kind`: to allow safe conversions within similar types.
  - `unsafe`: This is the default value, it allows any conversion, even if data might be lost.
- `subok`: If `True` (default), subclasses of the array will be preserved. If `False`, the result will always be a base-class array.
- `copy`: If `True` (default), a copy of the array is made. If `False`, reuse the input array if possible.
```

Reviewer: parameters are presented using ` ` backticks, and just reframed a few sentences

**Returns**

> A new array with the specified data type.

Reviewer: These two lines can be added in the introductory paragraph above

> ## Examples

Reviewer: ```suggestion
## Example
```

> 1. **Convert Float Array to Integer Array**

Reviewer: ```suggestion
The following example shows the use of the `.astype()` function to convert the elements of an array from float data type to integer data type:
```

   ```py

>    import numpy as np
> 
>    arr = np.array([1.2, 3.4, 5.6])
>    arr_int = arr.astype(int)
>    print(arr_int)

Reviewer: ```suggestion
import numpy as np

arr = np.array([1.2, 3.4, 5.6])
arr_int = arr.astype(int)
print(arr_int)
```

Reviewer: Python results in indentation errors due  to extra spacing

   ```

>    Output:

Reviewer: ```suggestion
The code above generates the following output:
```

>    ```py

Reviewer: ```suggestion
```shell
```

Reviewer: outputs are shown using the ```shell``` block

   [1 3 5]
   ```

> 2. **Convert Array to String Array**

Reviewer: ```suggestion
## Codebyte Example
```

Reviewer: the example 2 needs to be a codebyte example block, as mentioned in the issue

> 

Reviewer: ```suggestion

Run the following code to understand the use of `.astype()` function:

```

>    ```py

Reviewer: ```suggestion
```codebyte/python
```

>    arr = np.array([1, 2, 3])
>    arr_str = arr.astype(str)
>    print(arr_str)

Reviewer: ```suggestion
import numpy as np

arr = np.array([1, 2, 3])
arr_str = arr.astype(str)
print(arr_str)
```

Reviewer: indentation errors

   ```

>    Output:
> 
>    ```py
>    ['1' '2' '3']
>    ```
> 
> 

Reviewer: ```suggestion
```

Reviewer: A codebyte block should not be followed by an output block
